shared/middleware/helpers/bucket_helper.rb

```diff
@@ -271,8 +271,31 @@ def restore_previous_version(encrypted_channel_id, filename, version_id, user_id
     owner_id, channel_id = storage_decrypt_channel_id(encrypted_channel_id)
     key = s3_path owner_id, channel_id, filename
 
-    response = s3.copy_object(bucket: @bucket, key: key, copy_source: "#{@bucket}/#{key}?versionId=#{version_id}")
-
+    begin
+      response = s3.copy_object(
+        bucket: @bucket,
+        key: key,
+        copy_source: "#{@bucket}/#{key}?versionId=#{version_id}"
+      )
+    rescue
```

Comment:
to be sure we are not masking other errors, it seems like we should either specify that we are catching a NoSuchVersion error here (my recommendation), or else passing the exception we do catch as an argument to HoneyBadger.notify.

Reply:
Great idea, we'll do that.